Plugins/NativeEngine/Source/NativeEngine.cpp

```diff
@@ -711,19 +714,20 @@ namespace Babylon
                 bgfx::UniformInfo info{};
                 bgfx::getUniformInfo(uniforms[index], info);
                 auto itStage = uniformStages.find(info.name);
-                uniformInfos.emplace(std::make_pair<std::string, UniformInfo>(info.name, {itStage == uniformStages.end() ? uint8_t{} : itStage->second, uniforms[index]}));
+                auto& handle = uniforms[index];
+                uniformInfos.emplace(std::make_pair<std::string, UniformInfo>(info.name, {itStage == uniformStages.end() ? uint8_t{} : itStage->second, handle}));
+                uniformSizes[uniforms[index].idx] = static_cast<size_t>(info.num);
```

Comment:
I think you can store this info inside UniformInfo so that you don't need a second map?

Reply:
I think you would have to propagate the info to UniformValue and then it should work. Let's chat offline.

Reply:
Changed the code so that MaxElementSize is now part of UniformInfo struct.